Missing CMake dependency

https://travis-ci.org/github/boostorg/boost/jobs/769225032#L1226
```
Scanning dependencies of target boost_json
[ 31%] Building CXX object libs/json/CMakeFiles/boost_json.dir/src/src.cpp.o
In file included from /home/travis/build/boostorg/boost/libs/json/include/boost/json/detail/value_to.hpp:16:0,
                 from /home/travis/build/boostorg/boost/libs/json/include/boost/json/value_to.hpp:16,
                 from /home/travis/build/boostorg/boost/libs/json/include/boost/json.hpp:40,
                 from /home/travis/build/boostorg/boost/libs/json/include/boost/json/src.hpp:27,
                 from /home/travis/build/boostorg/boost/libs/json/src/src.cpp:10:
/home/travis/build/boostorg/boost/libs/json/include/boost/json/detail/index_sequence.hpp:15:44: fatal error: boost/mp11/integer_sequence.hpp: No such file or directory
```
Likely caused by `Boost::mp11` missing in the list of dependencies here: https://github.com/boostorg/json/blob/aae1863def3fb605d77ebbefc851cc26ac1758a9/CMakeLists.txt#L80-L89

`boostdep --cmake json` says
```
target_link_libraries(boost_json
  PUBLIC
    Boost::align
    Boost::assert
    Boost::config
    Boost::container
    Boost::exception
    Boost::mp11
    Boost::system
    Boost::throw_exception
    Boost::utility
)
```

--- a/.drone.star
+++ b/.drone.star
@@ -57,6 +57,7 @@ def main(ctx):
     linux_cxx("gcc 9 standalone", "g++-9", packages=" ".join(addon_gcc_9["apt"]["packages"]), image="ubuntu:18.04", buildtype="standalone", environment={"COMMENT": "standalone", "CXX": "g++-9"}),
     linux_cxx("gcc 10", "g++-10", packages=" ".join(addon_gcc_10["apt"]["packages"]), image="ubuntu:18.04", buildtype="boost", environment={"B2_TOOLSET": "gcc-10", "B2_CXXSTD": "17,2a"}),
     linux_cxx("gcc 10 standalone", "g++-10", packages=" ".join(addon_gcc_10["apt"]["packages"]), image="ubuntu:18.04", buildtype="standalone", environment={"COMMENT": "standalone", "CXX": "g++-10"}),
+    linux_cxx("gcc 10 cmake-superproject", "g++-10", packages=" ".join(addon_gcc_10["apt"]["packages"]), image="ubuntu:18.04", buildtype="cmake-superproject", environment={"COMMENT": "cmake-superproject", "CXX": "g++-10"}),
     linux_cxx("coverity", "", packages="", image="ubuntu:18.04", buildtype="coverity", environment={}),
     linux_cxx("docs", "", packages="docbook docbook-xml docbook-xsl xsltproc libsaxonhe-java default-jre-headless flex libfl-dev bison unzip rsync", image="cppalliance/droneubuntu1604:1", buildtype="docs", environment={"COMMENT": "docs"}),
     linux_cxx("codecov", "", packages=" ".join(addon_gcc_8["apt"]["packages"]), image="ubuntu:18.04", buildtype="codecov", environment={"COMMENT": "codecov.io","LCOV_BRANCH_COVERAGE": 0,"B2_CXXSTD": 11,"B2_TOOLSET": "gcc-8", "B2_DEFINES": "BOOST_NO_STRESS_TEST=1"}, stepenvironment={"CODECOV_TOKEN": {"from_secret": "codecov_token"} }),

--- a/.drone/cmake-superproject-script.sh
+++ b/.drone/cmake-superproject-script.sh
@@ -1,0 +1,50 @@
+#!/bin/bash
+
+set -ex
+
+echo '==================================> BEFORE_INSTALL'
+
+if  [ "$DRONE_BEFORE_INSTALL" = "gcc11" ]; then
+    wget http://kayari.org/gcc-latest/gcc-latest.deb
+    sudo dpkg -i gcc-latest.deb
+    export PATH=/opt/gcc-latest/bin:$PATH
+    export LD_RUN_PATH=/opt/gcc-latest/lib64
+fi
+
+echo '==================================> INSTALL'
+
+git clone https://github.com/boostorg/boost-ci.git boost-ci-cloned --depth 1
+cp -prf boost-ci-cloned/ci .
+rm -rf boost-ci-cloned
+# source ci/travis/install.sh
+# The contents of install.sh below:
+
+if [ "$DRONE_STAGE_OS" = "darwin" ]; then
+    unset -f cd
+fi
+
+export SELF=`basename $DRONE_REPO`
+export BOOST_CI_TARGET_BRANCH="$DRONE_COMMIT_BRANCH"
+export BOOST_CI_SRC_FOLDER=$(pwd)
+
+. ./ci/common_install.sh
+
+echo '==================================> COMPILE'
+
+export CXXFLAGS="-Wall -Wextra -Werror"
+
+mkdir __build_static
+cd __build_static
+cmake -DBOOST_ENABLE_CMAKE=1 -DBUILD_TESTING=ON -DBoost_VERBOSE=1 \
+    -DBOOST_INCLUDE_LIBRARIES=json ..
+cmake --build .
+ctest --output-on-failure -R boost_json
+
+cd ..
+
+mkdir __build_shared
+cd __build_shared
+cmake -DBOOST_ENABLE_CMAKE=1 -DBUILD_TESTING=ON -DBoost_VERBOSE=1 \
+    -DBOOST_INCLUDE_LIBRARIES=json -DBUILD_SHARED_LIBS=ON ..
+cmake --build .
+ctest --output-on-failure -R boost_json

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -84,6 +84,7 @@ elseif(BOOST_SUPERPROJECT_VERSION)
             Boost::config
             Boost::container
             Boost::exception
+            Boost::mp11
             Boost::system
             Boost::throw_exception
             Boost::utility

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -44,6 +44,7 @@ elseif(BOOST_SUPERPROJECT_VERSION)
             Boost::config
             Boost::container
             Boost::exception
+            Boost::mp11
             Boost::system
             Boost::throw_exception
             Boost::utility

